### PR TITLE
Make Nginx healthcheck script to tolerate unavailability

### PR DESCRIPTION
**Pull request by a-b**
* A short explanation of the proposed change:
Addresses https://github.com/cloudfoundry/capi-release/issues/364 health check will wait and retry if Nginx is not available.

* An explanation of the use cases your change solves
This change allows Nginx and webserver to start without monit interruption.

* [x] I have viewed signed and have submitted the Contributor License Agreement

* [x] I have made this pull request to the `develop` branch

* [x] I have run CF Acceptance Tests on bosh lite

### CODE REVIEW

File: jobs/cloud_controller_ng/templates/ccng_monit_http_healthcheck.sh.erb [hunk 1]
```diff
@@ -51,8 +51,8 @@ while true; do
   status=$?
   set -e
   if [[ $status > 0 ]] ; then
-    echo $(date --rfc-3339=ns) "ccng_monit_http_healthcheck failed to curl <${URL}>: exit code $status"
-    exit $status
+    echo "$(date --rfc-3339=ns) ccng_monit_http_healthcheck failed to curl <${URL}>: exit code $status"
+    [[ $status != 7 ]] && exit $status
```
Comment (Benjamintf1) [inline jobs/cloud_controller_ng/templates/ccng_monit_http_healthcheck.sh.erb:55]: I'd prefer a actual proper if statement, but maybe that's bikeshedding.

Comment (a-b) [reply jobs/cloud_controller_ng/templates/ccng_monit_http_healthcheck.sh.erb]: My preferred option is this syntax for one line, but it was changed to a classic if statement to improve readability.

**Comment by Gerg** [comment]
What is special about curl exit code 7? Would this same deadlock behavior not occur for other exit codes? It would be helpful to have details about this error case in the commit message.

Also, if we only care about the "Connection Refused" case, would it be more appropriate to use the `--retry-connrefused` curl flag?

**Comment by sethboyles** [comment]
I think there was hesitation to use `--retry-connrefused` as it was added to curl 7.52.0, and ubuntu-xenial stemcells (apparently) contain curl 7.47.0.

**Review by sethboyles — APPROVED** [review]

**Comment by a-b** [comment]
> What is special about curl exit code 7? Would this same deadlock behavior not occur for other curl exit codes? It would be helpful to have details about this error case in the commit message.
It was outlined in the https://github.com/cloudfoundry/capi-release/issues/364; I'll duplicate it in the commit message.

**Comment by a-b** [comment]
> What is special about curl exit code 7? Would this same deadlock behavior not occur for other curl exit codes? It would be helpful to have details about this error case in the commit message.

Updated commit message. 
Do you have any particular scenarios for [other curl exit codes](https://everything.curl.dev/usingcurl/returns) that you're worried about, @Gerg?

**Pull request merged**